mesonpy/_editable.py

```diff
@@ -2,183 +2,275 @@
-        return _NO_COLORS
-    elif 'FORCE_COLOR' in os.environ or sys.stdout.isatty():
-        return _COLORS
-    return _NO_COLORS
-
-
-_STYLES = _init_colors()  # holds the color values, should be _COLORS or _NO_COLORS
-
-
-class MesonpyFinder(importlib.abc.MetaPathFinder):
-    """Custom loader that whose purpose is to detect when the import system is
-    trying to load our modules, and trigger a rebuild. After triggering a
-    rebuild, we return None in find_spec, letting the normal finders pick up the
-    modules.
-    """
-
-    def __init__(
-        self,
-        project_name: str,
-        hook_name: str,
-        project_path: str,
-        build_path: str,
-        import_paths: List[str],
-        top_level_modules: List[str],
-        rebuild_commands: List[List[str]],
-        verbose: bool = False,
-    ) -> None:
-        self._project_name = project_name
-        self._hook_name = hook_name
-        self._project_path = project_path
-        self._build_path = build_path
-        self._import_paths = import_paths
-        self._top_level_modules = top_level_modules
-        self._rebuild_commands = rebuild_commands
-        self._verbose = verbose
+    class Traversable:
+        pass
+    class TraversableResources:
+        pass
 
-        for path in (self._project_path, self._build_path):
-            if not os.path.isdir(path):
-                raise ImportError(
-                    f'{path} is not a directory, but it is required to rebuild '
-                    f'"{self._project_name}", which is installed in editable '
-                    'mode. Please reinstall the project to get it back to '
-                    'working condition. If there are any issues uninstalling '
-                    'this installation, you can manually remove '
-                    f'{self._hook_name} and {os.path.basename(__file__)}, '
-                    f'located in {os.path.dirname(__file__)}.'
-                )
-
-    def __repr__(self) -> str:
-        return f'{self.__class__.__name__}({self._project_path})'
-
-    def _debug(self, msg: str) -> None:
-        if self._verbose:
-            print(msg.format(**_STYLES))
-
-    def _proc(self, command: List[str]) -> None:
-        # skip editable hook installation in subprocesses, as during the build
-        # commands the module we are rebuilding might be imported, causing a
-        # rebuild loop
-        # see https://github.com/mesonbuild/meson-python/pull/87#issuecomment-1342548894
-        env = os.environ.copy()
-        env['_MESONPY_EDITABLE_SKIP'] = os.pathsep.join((
-            env.get('_MESONPY_EDITABLE_SKIP', ''),
-            self._project_path,
-        ))
 
-        if self._verbose:
-            subprocess.check_call(command, cwd=self._build_path, env=env)
-        else:
-            subprocess.check_output(command, cwd=self._build_path, env=env)
+MARKER = 'MESON_PYTHON_EDITABLE_SKIP'
+VERBOSE = 'MESON_PYTHON_EDITABLE_VERBOSE'
```

Comment:
You should rename this to ``MESONPY_EDITABLE_VERBOSE`` or fix the docs.

Reply:
The project name is meson-python, thus I think having things named `mesonpy` is a mistake. I'll update the documentation.

Reply:
I renamed the variables to have the `MESONPY_` prefix. I still think it would be nicer to use the `MESON_PYTHON_` prefix, but we can have that discussion separately.